lib/rules/no-restricted-imports.js

```diff
@@ -107,40 +116,66 @@ module.exports = {
             return memo;
         }, {});
 
-        // if no imports are restricted we don"t need to check
-        if (Object.keys(restrictedPaths).length === 0 && restrictedPatterns.length === 0) {
-            return {};
-        }
-
         const restrictedPatternsMatcher = ignore().add(restrictedPatterns);
 
-        /**
-         * Checks to see if "*" is being used to import everything.
-         * @param {Set.<string>} importNames Set of import names that are being imported
-         * @returns {boolean} whether everything is imported or not
-         */
-        function isEverythingImported(importNames) {
-            return importNames.has("*");
-        }
-
         /**
          * Report a restricted path.
+         * @param {string} importSource path of the import
+         * @param {Map<string:Object} importNames Map of import names that are being imported
          * @param {node} node representing the restricted path reference
          * @returns {void}
          * @private
          */
-        function reportPath(node) {
-            const importSource = node.source.value.trim();
-            const customMessage = restrictedPathMessages[importSource] && restrictedPathMessages[importSource].message;
+        function reportPath(importSource, importNames, node) {
+            if (!restrictedPathMessages[importSource]) {
+                return;
+            }
 
-            context.report({
-                node,
-                messageId: customMessage ? "pathWithCustomMessage" : "path",
-                data: {
-                    importSource,
-                    customMessage
-                }
-            });
+            const customMessage = restrictedPathMessages[importSource].message;
+            const restrictedImportNames = restrictedPathMessages[importSource].importNames;
+
+            if (importNames.has("*")) {
```

Comment:
Should this be inside ` if (restrictedImportNames)`?

If the whole module is restricted (not just some `importNames` from the module, i.e. configuration for that module doesn't have `importNames`), it will be reported below with `"path"` or `"pathWithCustomMessage"` messageId.

As it is now in this PR, I think it would report two messages: one for the whole module (which is correct) and another for `*` (which shouldn't be reported when the whole module is restricted; also, the message for `*` expects `importNames` which doesn't exists in this case). It would be nice to add a test for this.

Reply:
Good catch! You're totally right 👍

Reply:
Done!